### PR TITLE
Opening a subVI will show the project item in the files pane

### PR DESCRIPTION
**Pull request by abahetik**
When a subVI is opened the file explorer/ file pane does not reflect it. So in this feature we want the file pane to highlight the subVI location. 

So, there can be two cases:-
1. If the node corresponding to the path is not created
- We create the sub-tree for the file path and expand the tree along the path
- We use focusNode() to unset the previous VI selection and then set the current subVI as the selected VI.
2. If the node has already been created
- We expand the tree if it is collapsed.
- Then we use focusNode() to unset the previous VI selection and then set the current subVI as the selected VI.

Note: The entry point is window.onLoad() in wrapper.ts in order to understand the the workflow.

### CODE REVIEW

File: src/script/treeView/lib/aimaratree.ts [hunk 1]
```diff
@@ -1,5 +1,3 @@
-// This code is based on aimaraJS - Pure Javascript TreeView Component (https://github.com/rafaelthca/aimaraJS).
```
Comment (pratheekshasn) [inline src/script/treeView/lib/aimaratree.ts:1]: This comment should not be removed. Please undo this.

File: src/script/treeView/lib/wrapper.ts [hunk 2]
```diff
@@ -111,30 +111,63 @@ function updateTreeViewAndSearchResults(searchStr: string) {
...
+        document.getElementsByClassName("node_selected")[0].className = "node";
+        focusNode.className = "node_selected";
     }
-    const selectedNode = document.getElementsByClassName("node_selected");
-    if (selectedNode.length > 0) {
-        selectedNode[0].className = "node";
+}
+function expandUpwardTree(decodedJsonPath) {
+    const pathList = decodedJsonPath.split("/");
+    // tslint:disable-next-line:prefer-const
```
[the remark this answers is not in the document]
Comment (abahetik) [reply src/script/treeView/lib/wrapper.ts]: Done

File: src/script/treeView/lib/wrapper.ts [hunk 3]
```diff
@@ -111,30 +111,63 @@ function updateTreeViewAndSearchResults(searchStr: string) {
...
-    const nodeToSelect = document.getElementById(jsonPath).parentNode;
-    (nodeToSelect.childNodes[1] as HTMLSpanElement).className = "node_selected";
-    let nodeToExpand = document.getElementById(jsonPath);
-    while (nodeToExpand.id !== escape(rootNode.path)) {
-        nodeToExpand = (nodeToExpand.parentNode.parentNode) as HTMLUListElement;
-        (nodeToExpand as HTMLUListElement).style.display = "block";
-        const nodeExpColImg = nodeToExpand.getElementsByTagName("img")[0];
-        nodeExpColImg.id = "toggle_off";
-        nodeExpColImg.src = "images/collapse.png";
+    const selectNodeParent = document.getElementById(decodeURI(decodedJsonPath));
```
[the remark this answers is not in the document]
Comment (abahetik) [reply src/script/treeView/lib/wrapper.ts]: Done

File: src/script/treeView/lib/aimaratree.ts [hunk 4]
```diff
@@ -433,7 +432,7 @@ export class TreeHelper {
         return parentObj;
     }
 
-    public givenTreeExpandAtTopLevel(treeType) {
+    public getTreeAndCallDblEvent(treeType) {
```
[the remark this answers is not in the document]
Comment (abahetik) [reply src/script/treeView/lib/aimaratree.ts]: Okay

File: src/script/treeView/lib/aimaratree.ts (hunk 1 above)
Comment (abahetik) [reply src/script/treeView/lib/aimaratree.ts:1]: Added back

File: src/script/treeView/lib/wrapper.ts [hunk 5]
```diff
@@ -111,30 +111,65 @@ function updateTreeViewAndSearchResults(searchStr: string) {
...
+        document.getElementsByClassName("node_selected")[0].className = "node";
+    } else {
+        const focusNode = document.getElementById(decodedJsonPath);
+        let focusNodeTemp = focusNode;
+        /* If the tree strcuture for the file that we request is not expanded then,
+        we expand it first. Then, we highlight(node_selected) the file in the file-pane.
+        Else, we just do the node selection in the file pane for the requested file,
+        as the file tree is already open */
+        if (focusNodeTemp == null) {
+            expandUpwardTree(decodedJsonPath);
```
Comment (pratheekshasn) [inline src/script/treeView/lib/wrapper.ts:152]: You could consider renaming `expandUpwardTree()` to `createSubTreeAndExpandBranchForPath()`. And you can extract the code in the else block into `expandBranchForPath()`.

File: src/script/treeView/lib/wrapper.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (pratheekshasn) [reply src/script/treeView/lib/wrapper.ts]: Just out of curiosity, how come tslint no longer complains about the for loop? That also had a tslint-disable on it a few commits ago, I'm curious to know why it doesn't complain about the same anymore. Have you run tslint on this code?

File: src/script/treeView/lib/wrapper.ts [hunk 6]
```diff
@@ -111,30 +111,63 @@ function updateTreeViewAndSearchResults(searchStr: string) {
...
-        nodeExpColImg.src = "images/collapse.png";
+    const selectNodeParent = document.getElementById(decodeURI(decodedJsonPath));
+    const selectedNode = selectNodeParent.getElementsByTagName("span")[0];
+    selectedNode.scrollIntoView({ behavior: "smooth", block: "nearest", inline: "start" });
+    focusGivenNode(selectedNode);
+}
+
+export function toggleSelectedNode(jsonPath) {
+    const decodedJsonPath = decodeURIComponent(jsonPath);
+    if (decodedJsonPath.split("/").pop() === "control") {
```
[the remark this answers is not in the document]
Comment (pratheekshasn) [reply src/script/treeView/lib/wrapper.ts:160]: You can leave this as is for now if it takes too long to investigate, we can always do that later.

[the remark this answers is not in the document]
Comment (abahetik) [reply src/script/treeView/lib/wrapper.ts:160]: Ok, @sushmitha will be able to tell the reason for it better.

File: src/script/treeView/lib/wrapper.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (abahetik) [reply src/script/treeView/lib/wrapper.ts]: The tslint is enabled but I am not sure why it didnot complain.

**Pull request merged**